### PR TITLE
Author Crud

### LINKED ISSUE #4: Implement CRUD for Author
System operations that need to be implemented:

- **Author search**
- **Add author**
- **Update author**
- **Delete author**

### PR DESCRIPTION
**Pull request by VladHolobyn**
Closes #4

### CODE REVIEW

File: src/main/java/com/pnudev/librarysystem/controller/AuthorController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @Value("${application.defaultPageSize}")
+    private int defaultPageSize;
+
+    private final AuthorService authorService;
+
+    @GetMapping
+    public Page<AuthorDTO> searchAuthor(
+            @RequestParam("firstname") Optional<String> firstName,
+            @RequestParam("lastname") Optional<String> lastName,
+            @RequestParam("page") Optional<Integer> pageNumber,
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnudev/librarysystem/controller/AuthorController.java]: pageable

File: src/main/java/com/pnudev/librarysystem/service/AuthorService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+@AllArgsConstructor
+@Service
+public class AuthorService {
+
+    private final AuthorRepository authorRepository;
+    private final AuthorMapper authorMapper;
+
+    public AuthorDTO addAuthor(AuthorDTO authorDTO) {
+        return authorMapper.toDTO(authorRepository.save(authorMapper.toEntity(authorDTO)));
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnudev/librarysystem/service/AuthorService.java]: three lines

File: src/main/java/com/pnudev/librarysystem/service/AuthorService.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        Author author = authorRepository.findById(id)
+                .orElseThrow(()-> new EntityNotFoundException("Author with id: %d not found".formatted(id)));
+
+        authorMapper.updateAuthorFromDTO(authorDTO,author);
+
+        return authorMapper.toDTO(authorRepository.save(author));
+    }
+
+    public Page<AuthorDTO> searchAuthorByParams(
+            Optional<String> firstName,
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnudev/librarysystem/service/AuthorService.java]: Optional is not used as a param type

File: src/main/java/com/pnudev/librarysystem/specifcation/AuthorSpecifications.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.pnudev.librarysystem.specifcation;
+
+import com.pnudev.librarysystem.entity.Author;
+import org.springframework.data.jpa.domain.Specification;
+
+public class AuthorSpecifications {
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnudev/librarysystem/specifcation/AuthorSpecifications.java]: SpecificationUtils

File: src/main/resources/application.yaml [hunk 5]
```diff
@@ -6,3 +6,6 @@ spring:
   jpa:
     hibernate:
       ddl-auto: validate
+application:
```
Comment (Igor-Polatajko) [inline src/main/resources/application.yaml]: just in code (Pageable)

File: src/main/java/com/pnudev/librarysystem/controller/AuthorController.java (hunk 1 above)
Comment (VladHolobyn) [reply src/main/java/com/pnudev/librarysystem/controller/AuthorController.java]: Done

File: src/main/java/com/pnudev/librarysystem/service/AuthorService.java (hunk 2 above)
Comment (VladHolobyn) [reply src/main/java/com/pnudev/librarysystem/service/AuthorService.java]: Done

File: src/main/java/com/pnudev/librarysystem/service/AuthorService.java (hunk 3 above)
Comment (VladHolobyn) [reply src/main/java/com/pnudev/librarysystem/service/AuthorService.java]: Refactored

File: src/main/java/com/pnudev/librarysystem/specifcation/AuthorSpecifications.java (hunk 4 above)
Comment (VladHolobyn) [reply src/main/java/com/pnudev/librarysystem/specifcation/AuthorSpecifications.java]: Done

File: src/main/resources/application.yaml (hunk 5 above)
Comment (VladHolobyn) [reply src/main/resources/application.yaml]: Refactored

File: src/main/java/com/pnudev/librarysystem/util/SpecificationUtils.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.pnudev.librarysystem.util;
+
+import org.springframework.data.jpa.domain.Specification;
+
+public class SpecificationUtils{
```
Comment (justme1135) [inline src/main/java/com/pnudev/librarysystem/util/SpecificationUtils.java:7]: Use Lombok's `@UtilityClass` for this

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: src/main/java/com/pnudev/librarysystem/util/SpecificationUtils.java (hunk 6 above)
Comment (VladHolobyn) [reply src/main/java/com/pnudev/librarysystem/util/SpecificationUtils.java:7]: Done

**Review by justme1135 — APPROVED** [review]

**Pull request merged**